helix-term/src/ui/explorer.rs

```diff
@@ -0,0 +1,1464 @@
+            let path = parent.to_path_buf();
+            self.change_root(path)
+        } else {
+            Ok(())
+        }
+    }
+
+    pub fn is_opened(&self) -> bool {
+        self.state.open
+    }
+
+    pub fn column_width(&self) -> u16 {
+        self.column_width
+    }
+
+    fn increase_size(&mut self) {
+        const EDITOR_MIN_WIDTH: u16 = 10;
+        self.column_width = std::cmp::min(
+            self.state.area_width.saturating_sub(EDITOR_MIN_WIDTH),
+            self.column_width.saturating_add(1),
+        )
+    }
+
+    fn decrease_size(&mut self) {
+        self.column_width = self.column_width.saturating_sub(1)
+    }
+
+    fn rename_current(&mut self, line: &String) -> Result<()> {
+        let item = self.tree.current_item()?;
+        let path = PathBuf::from(line);
+        if let Some(parent) = path.parent() {
+            std::fs::create_dir_all(parent)?;
+        }
+        std::fs::rename(&item.path, &path)?;
+        self.tree.refresh()?;
+        self.reveal_file(path)
+    }
+
+    fn remove_folder(&mut self) -> Result<()> {
+        let item = self.tree.current_item()?;
+        std::fs::remove_dir_all(&item.path)?;
+        self.tree.refresh()
+    }
+
+    fn remove_file(&mut self) -> Result<()> {
+        let item = self.tree.current_item()?;
+        std::fs::remove_file(&item.path)?;
+        self.tree.refresh()
+    }
+}
+
+fn close_documents(current_item_path: PathBuf, cx: &mut Context) -> Result<()> {
+    let ids = cx
+        .editor
+        .documents
+        .iter()
+        .filter_map(|(id, doc)| {
+            if doc.path()?.starts_with(&current_item_path) {
+                Some(*id)
+            } else {
+                None
+            }
+        })
+        .collect::<Vec<_>>();
+
+    for id in ids {
+        cx.editor.close_document(id, true)?;
+    }
+    Ok(())
+}
+
+impl Component for Explorer {
+    /// Process input events, return true if handled.
+    fn handle_event(&mut self, event: &Event, cx: &mut Context) -> EventResult {
+        if self.tree.prompting() {
+            return self.tree.handle_event(event, cx, &mut self.state);
+        }
+        let key_event = match event {
+            Event::Key(event) => event,
+            Event::Resize(..) => return EventResult::Consumed(None),
+            _ => return EventResult::Ignored(None),
+        };
+        if !self.is_focus() {
+            return EventResult::Ignored(None);
+        }
+        if let Some(mut on_next_key) = self.on_next_key.take() {
+            return on_next_key(cx, self, key_event);
+        }
+
+        if let EventResult::Consumed(c) = self.handle_prompt_event(key_event, cx) {
+            return EventResult::Consumed(c);
+        }
+
+        (|| -> Result<()> {
+            match key_event {
+                key!(Esc) => self.unfocus(),
+                key!('q') => self.close(),
+                key!('?') => self.toggle_help(),
```

Comment:
```suggestion
                key!('?') | shift!('?') => self.toggle_help(),
```
`shift + ?` is the key event you will get on windows for most keyboards. Also, tested it in the current state doesn't do anything on hitting `shift + /` `==>` `shift + ?`.